bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -322,6 +396,30 @@ private void scan() {
         logger.trace("IpAddonFinder::scan completed");
     }
 
+    // build from plaintext string
+    private byte[] buildRequestArrayPlain(DatagramChannel channel, String request)
+            throws java.io.IOException, ParseException {
+        InetSocketAddress sock = (InetSocketAddress) channel.getLocalAddress();
+
+        // replace first
+        StringBuffer req = new StringBuffer(request);
```

Comment:
The `StringBuilder` is preferred when there is no need for synchronization as it has better performance:

```suggestion
        StringBuilder req = new StringBuilder(request);
```